### PR TITLE
Raise PHPStan level to 8

### PR DESCRIPTION
**Pull request by derrabus**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | N/A

#### Summary

Let's make our static analysis a little bit stricter.

### CODE REVIEW

File: src/DependencyFactory.php [hunk 1]
```diff
@@ -145,7 +145,7 @@ public function getConfiguration(): Configuration
     {
         if ($this->configuration === null) {
             $this->configuration = $this->configurationLoader->getConfiguration();
-            $this->freeze();
+            $this->frozen = true;
```
Comment (SenseException) [inline src/DependencyFactory.php:148]: Why these changes where the `frozen()` method gets replaced with the direct set to true?

Comment (derrabus) [reply src/DependencyFactory.php:148]: PHPStan treats the method calls as impure and complains that `$this->configuration` might be `null` when we return it.

Comment (SenseException) [reply src/DependencyFactory.php:148]: I was playing around with annotations and other configurations, but it seems like PHPStan can't be convinced that `$this->freeze()` does nothing evil with other property states.

**Review by SenseException — APPROVED** [review]

File: src/DependencyFactory.php (hunk 1 above)
Comment (derrabus) [reply src/DependencyFactory.php:148]: Yes, and this is fine imho. `freeze()` is impure, so PHPStan's behavior is correct.
